### ISSUE TITLE
Directly use Zed lake query endpoint

### ISSUE THREAD

**Issue by philrz** [brimdata/brim#1618]
As described in brimdata/zed#2680, the current API is allowing Brim to access data in pools via the legacy "search" endpoint, but there's functionality such as `join` from multiple pools that will only be possible when the app starts directly using the new Zed lake "query" endpoint.

**Comment by philrz** [comment]
I took a shot at verifying the work thus far with Brim 0dcf10c by attempting this multi-pool `join` derived from the [Zed docs](https://github.com/brimdata/zed/tree/main/docs/language/operators#example-1---inner-join):

```
from (
  fruit.ndjson => sort flavor;
  people.ndjson => sort likes;
) | inner join on flavor=likes eater:=name
```

This produced a backend error `Upstream data source blocked by 'from operator'`. I could see in Wireshark that the query being sent to the backend was:

```
{"query":"from '0x0dcfc31bd80817b052e0c8eb90a8297af928a8ea' | from (   fruit.ndjson => sort flavor;   people.ndjson => sort likes; ) | inner join on flavor=likes eater:=name | head 10000"}
```

@nwt proposed a fix for this:

> Rather than prepend a `from` to the query, the app should set the head field in the request, like this.
> 
>```
>{
>  "query": "from (   fruit.ndjson => sort flavor;   people.ndjson => sort likes; ) | inner join on flavor=likes eater:=name | head 10000",
>  "head": "0x0dcfc31bd80817b052e0c8eb90a8297af928a8ea"
>}
>```

**Comment by philrz** [comment]
Verified in Brim commit de44850.

Now with the benefit of the additional fix from #1836, we can execute the [multi-pool join shown in the Zed docs](https://github.com/brimdata/zed/tree/main/docs/language/operators#example-1---inner-join) from within the Brim app, as shown in the attached video:

https://user-images.githubusercontent.com/5934157/133686590-562cde0c-5afd-4f22-b2da-c337c0d7b0f2.mp4

In a discussion with the Dev team, it was confirmed that the current fix added in #1836 is only limited to queries where the `from` leads off the query entered by the user in the search box. That is, queries with `from` nested somewhere deeper inside the Zed are not expected to work yet.

Thanks @mason-fish!
